packages/webapp/src/components/Form/Unit/useUnit.jsx

```diff
@@ -0,0 +1,212 @@
+/*
+ *  Copyright 2019, 2020, 2021, 2022, 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { useCallback, useEffect, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { get, useFormState } from 'react-hook-form';
+import {
+  area_total_area,
+  getDefaultUnit,
+  roundToTwoDecimal,
+} from '../../../util/convert-units/unit';
+import { convert } from '../../../util/convert-units/convert';
+import { getUnitOptionMap } from '../../../util/convert-units/getUnitOptionMap';
+
+const getOptions = (unitType = area_total_area, system) => {
+  return unitType[system].units.map((unit) => getUnitOptionMap()[unit]);
+};
+
+const useUnit = ({
+  disabled,
+  name,
+  displayUnitName,
+  hookFormSetValue,
+  hookFormGetValue,
+  hookFromWatch,
+  defaultValue,
+  system,
+  control,
+  unitType,
+  from: defaultValueUnit,
+  to,
+  required,
+  optional,
+  mode,
+  max,
+}) => {
+  const onClear = () => {
+    setVisibleInputValue('');
+    hookFormSetHiddenValue('', { shouldClearError: !optional, shouldValidate: optional });
+  };
+
+  const [showError, setShowError] = useState(false);
+  const [isDirty, setDirty] = useState(false);
+  const { errors } = useFormState({ control });
+  const error = get(errors, name);
+
+  useEffect(() => {
+    setShowError(!!error && !disabled && isDirty);
+  }, [error]);
+
+  const { displayUnit, displayValue, options, databaseUnit, isSelectDisabled, measure } =
+    useMemo(() => {
+      const databaseUnit = defaultValueUnit ?? unitType.databaseUnit;
+      const options = getOptions(unitType, system);
+      const hookFormValue = hookFormGetValue(name);
+      const value = hookFormValue || (hookFormValue === 0 ? 0 : defaultValue);
+      const isSelectDisabled = options.length <= 1;
+      const measure = convert().describe(databaseUnit)?.measure;
+      return to && convert().describe(to)?.system === system
+        ? {
+            displayUnit: to,
+            displayValue:
+              defaultValue && roundToTwoDecimal(convert(value).from(databaseUnit).to(to)),
+            options,
+            databaseUnit,
+            isSelectDisabled,
+            measure,
+          }
+        : {
+            ...getDefaultUnit(unitType, value, system, databaseUnit),
```

Comment:
This conditional branch is only used in location pages. Not sure if it's better to extract this to somewhere else.